File: frontend/lib/models/diet_plan.dart
import 'dart:convert';

DietPlan dietPlanFromJson(String str) => DietPlan.fromJson(json.decode(str));

String dietPlanToJson(DietPlan data) => json.encode(data.toJson());

class DietPlan {
  DietPlan({
    this.caloriePerDay,
    this.dietId,
    this.endDate,
    this.id,
    this.startDate,
    this.userId,
    this.status,
    this.weightNow,
    this.weightTarget,
  });

  double caloriePerDay;
  String dietId;
  String endDate;
  int id;
  String startDate;
  int userId;
  int status;
  double weightNow;
  double weightTarget;

  factory DietPlan.fromJson(Map<String, dynamic> json) => DietPlan(
        caloriePerDay: json["caloriePerDay"],
        dietId: json["dietId"],
        endDate: json["endDate"],
        id: json["id"],
        startDate: json["startDate"],
        userId: json["userId"],
        weightNow: json["weightNow"],
        weightTarget: json["weightTarget"],
        status: json['status'],
      );

  Map<String, dynamic> toJson() => {
        "caloriePerDay": caloriePerDay,
        "dietId": dietId,
        "endDate": endDate,
        "id": id,
        "startDate": startDate,
        "userId": userId,
        "weightNow": weightNow,
        "weightTarget": weightTarget,
        "status": status
      };
}

File: backend/src/main/java/com/fitness/backend/models/DietPlan.java
package com.fitness.backend.models;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class DietPlan {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String startDate;
    private String endDate;
    private double weightNow;
    private double weightTarget;
    private double caloriePerDay;
    private String dietId;
    private int status; // 1 = active, 0 inactive
    private int userId;
}
